internal/cache/devices.go

```diff
@@ -92,7 +92,7 @@ func (d *deviceCache) ForName(name string) (models.Device, bool) {
 	if !ok {
 		return models.Device{}, false
 	}
-	return *device, ok
+	return device.Clone(), ok
```

Comment:
We can add a comment to remind why we return clone here.
```suggestion
        // As the cache device contains pointer fields(map and slice), directly return device may cause concurrent map read 
	// or write if the invoker spawn another goroutine to manipulate the device, so returning the clone of device here
	return device.Clone(), ok
```

Reply:
Added the comment.